src/commands/createAuthorizationPermission.ts

```diff
@@ -0,0 +1,147 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { HttpOperationResponse, ServiceClient } from "@azure/ms-rest-js";
+import { ProgressLocation, QuickPickItem, window } from "vscode";
+import { createGenericClient, IActionContext } from "vscode-azureextensionui";
+import { ApimService } from "../azure/apim/ApimService";
+import { AuthorizationPermissionsTreeItem, IAuthorizationPermissionTreeItemContext } from "../explorer/AuthorizationPermissionsTreeItem";
+import { AuthorizationTreeItem } from "../explorer/AuthorizationTreeItem";
+import { ext } from "../extensionVariables";
+import { localize } from "../localize";
+
+const otherManagedIdentitiesOptionLabel = "Other managed identities..."
+const customOptionLabel = "Custom...";
+
+export async function createAuthorizationPermission(context: IActionContext & Partial<IAuthorizationPermissionTreeItemContext>, node?: AuthorizationPermissionsTreeItem): Promise<void> {
+    if (!node) {
+        const AuthorizationNode = <AuthorizationTreeItem>await ext.tree.showTreeItemPicker(AuthorizationTreeItem.contextValue, context);
+        node = AuthorizationNode.authorizationPermissionsTreeItem;
+    }
+
+    const apimService = new ApimService(node.root.credentials, node.root.environment.resourceManagerEndpointUrl, node.root.subscriptionId, node.root.resourceGroupName, node.root.serviceName);
+
+    const identityOptions = await populateIdentityOptionsAsync(
+        apimService, node.root.credentials, node.root.environment.resourceManagerEndpointUrl);
+    const identitySelected = await ext.ui.showQuickPick(
+        identityOptions, { placeHolder: 'Select Identity...', canPickMany: false, suppressPersistence: true });
+
+    if (identitySelected.label == otherManagedIdentitiesOptionLabel) {
+        var otherManagedIdentityOptions = await populateOtherManageIdentityOptions(node.root.credentials, node.root.environment.resourceManagerEndpointUrl, node.root.subscriptionId);
+        var managedIdentitySelected = await ext.ui.showQuickPick(
+            otherManagedIdentityOptions, { placeHolder: 'Select Managed Identity ...', canPickMany: false });
+        var permissionName = managedIdentitySelected.label;
+        var oid = managedIdentitySelected.description!;
+    }
+    else if (identitySelected.label == customOptionLabel) {
+        // No object id specified; ask explicitly
+        var permissionName = await askInput('Enter Permission name ...');
+        var oid = await askInput('Enter Object Id ...');
```

Comment:
you can leave some help text with how to get it from azure portal or az cli. probably link to our github repo with help doc might also work.